src/GraphQL.MicrosoftDI/ScopedConnectionBuilderExtensions.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Threading.Tasks;
+using GraphQL.Builders;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.MicrosoftDI
+{
+    /// <summary>
+    /// Extension methods for creating connection resolver builders.
+    /// </summary>
+    public static class ScopedConnectionBuilderExtensions
+    {
+        /// <summary>
+        /// Sets the resolver for the connection field. A dependency injection scope is created for the duration of the resolver's execution
+        /// and the scoped service provider is passed within <see cref="IResolveFieldContext.RequestServices"/>.
+        /// </summary>
+        public static void ResolveScoped<TSourceType, TReturnType>(this ConnectionBuilder<TSourceType> builder, Func<IResolveConnectionContext<TSourceType>, TReturnType> resolver)
```

Comment:
For corresponding methods for FieldBuilder we have `ScopedFieldResolver` and `ScopedAsyncFieldResolver`. Why there are no such?

Reply:
There did not seem to be a reason to add them.  Since `IFieldResolver.Resolve` only accepts `IResolveFieldContext`, the connection context would need to be built within the `ScopedConnectionResolver`.  And I can't access the private `PageSize` or `Unidirectional` properties of the builder to be able to build a new connection context.  And people cannot use `.Resolve(IFieldResolver)` on a connection; they would need to access the underlying `FieldType`.

And if someone actually does want to use their own `IFieldResolver` for a connection context, they can write this:

```cs
var resolver = new ScopedFieldResolver<int>(context =>
{
    var connectionContext = new GraphQL.Builders.ResolveConnectionContext<object>(context, isUnidirectional: false, defaultPageSize: 100);
    //code here
});
```
just about as easily as this:
```cs
var resolver = new ScopedConnectionResolver<int>(connectionContext =>
{
    //code here
}, isUnidirectional: false, defaultPageSize: 100);
```

In short, I felt this code is more for the connection builders rather than fields in general, and since it is not needed by the builders, I did not feel the need to add them.  I can add them if you think I should -- but again, they cannot be used by the connection builders without additional changes to the codebase.

Reply:
IMO
1. `ConnectionBuilder` should expose 3 properties that now are private fields: _isUnidirectional, _isBidirectional, _pageSize
2. `ResolveScoped` and `ResolveScopedAsync` should return `ConnectionBuilder<>`, not void.

Reply:
We have reviewed and addressed your suggested changes within #2458 

I have carefully revisited the idea of adding a `ScopedConnectionResolver`.  There still does not appear to be any reason to add this class.  While we can pull the page size from metadata and infer the `Bidirectional` property from `FieldType.Arguments`, there is no matching `ConnectionResolver` in the base project.  Also there is no `.Resolve(IFieldResolver resolver)` method exposed on the connection builder (although `FieldType` is exposed as a workaround).  And finally, it is very easy, as described above, to implement your own connection resolver using the `ScopedFieldResolver` method in conjunction with constructing a new `ResolveConnectionContext`.  So for those reasons I still am not adding a `ScopedConnectionResolver` class in this PR.

The only benefit would occur if we ALSO integrate `ScopedResolveConnectionContextAdapter` (an internal class) with `ResolveConnectionContext<T>`.  This would result in one less allocation per execution, at the cost of a more complex code design.  Implementing either one without the other does not remove the allocation.

Would you agree with this thought process?

Reply:
I don't know yet. I need to see. Recently, I have less time to work on the project, so the review is delayed.

Reply:
gentle reminder @sungam3r

Reply:
I did not read  email for a few days, was busy. I will do the review tomorrow. As I remember, we have already closed most questions.

Reply:
Why did you release 4.3 without this? I thought you were going to review it today? It’s all done.

Reply:
Now we’d need to release 4.4 on the same day as 4.3

Reply:
I forgot. I do the review right now.